src/components/PopupWithForm.js

```diff
@@ -36,4 +38,14 @@ export default class PopupWithForm extends Popup {
     });
     super.setEventListeners();
   }
+
+  setButtonText(isSaving) {
+    if (isSaving) {
+      this._submitButtonText = "Saving...";
+      console.log(this._submitButtonText);
+    } else {
+      this._submitButtonText = this._submitButton.textContent;
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```